test/set.test.js

```diff
@@ -61,7 +68,7 @@ test('Set forEach', () => {
 test('Set iterators', () => {
   let a = newSet();
   let values = a.values();
-  expect(values.next().value).toBe(a.first);
+  expect(values.first).toBe(a.first);
```

Comment:
This change doesn't make sense, because the whole purpose of this test is to check the iterator `.next()`.

Please revert this change. Same below.

Reply:
I was confused as to why the tests were different...

Reply:
That is deliberate. It tests that the results from various methods are consistent and as expected. If you test the function against itself, it will return the same wrong result twice and pass (false positive).

In any case, the `.next().value` is what this test is really testing, so this needs to stay.

Reply:
... and indeed why the SetColl returns an iterator while the other collections return a collection...

Reply:
The methods weren't consistent across collections, which is why I made the original change. I'm not testing the function against itself (ironically your version in `test/array.test.js` which I coped from comes closest to testing the function against itself, as the only difference is that the array is a copy).